README.md

```diff
@@ -63,27 +63,36 @@ Ensure you have the following installed:
  * [Rust & Cargo](https://www.rust-lang.org/tools/install)
 
 ## Setup
-1. Create a working directory eg(cd didcomm-mediator-rs) and cd into your directory.
+1. (Optional) Create a working directory eg(cd didcomm-mediator-rs) and cd into your directory.
 
  2. Clone the repository using the following command:
-
-git clone https://github.com/adorsys/didcomm-mediator-rs.git
+```sh
+git clone git@github.com:adorsys/didcomm-mediator-rs.git
+```
 
 ## Troubleshooting Tips
 If you encounter any issues while running the application, here are some troubleshooting tips to help you resolve them
  
  ### Common Issues
 
-  1. Build Errors:
+   Build Errors:
   * Ensure that you have the required system packages installed. `libssl-dev` and `pkg-config` 
   * Ensure that you have the latest version of Rust and Cargo installed.
-  * CHeck for any missing dependencies using `cargo check`.
+  * Check for any missing dependencies using 
+```sh
+cargo check
+```
 
 ## Example
- 1. Start the mediator service:
-
-cd didcomm-mediator-rs/mediator
-cargo run
-
+ Starting the mediator service:
+ 
+Since the didcomm-mediator-rs is made up of so many `crates` both `libraries` and `binaries`, to run anything you have to explicitly specify the `crate` and the `binary` you want to run like for example
+```sh
+cargo run -p <package> --bin <binary>
+```
```

Comment:
Disregard `mediator-server`. The only binary crate is `generic-server`, hence the only crate you can run. You may launch the generic server by selecting some features mapping to the plugins you would want to be loaded. Can you sync with @Christiantyemele please? There is also the important step of configuring env variables that he can support you in documenting.

Reply:
@Christiantyemele please check this message.

Reply:
Okay sir

Reply:
> Disregard `mediator-server`. The only binary crate is `generic-server`, hence the only crate you can run. You may launch the generic server by selecting some features mapping to the plugins you would want to be loaded. Can you sync with @Christiantyemele please? There is also the important step of configuring env variables that he can support you in documenting.

Ok, :eyes:

Reply:
@Blindspot22 - This is no longer blocked. Are you proceeding with it?

Reply:
You can check again and let me know if more suggestions!

Reply:
Can you update the branch so we have `readme.png` here?

Reply:
Okay sir. Done, I think the image is clear now.